examples/x-to-zkevm-migration-app/frontend/src/app/page.tsx

```diff
@@ -0,0 +1,209 @@
+          setIMXAssets(assetsResponse.result);
+        } else {
+          console.log('IMX wallet address is undefined');
+        }
+      } catch (error) {
+        console.error('Error listing IMX assets:', error);
+      } finally {
+        setLoading(false);
+      }
+    }
+  };
+
+  const handleListZKEVMAssets = async () => {
+    if (listZKEVMAssets) {
+      setLoading(true);
+      try {
+        if (imxWalletAddress) {
+          const assetsResponse = await listZKEVMAssets(imxWalletAddress);
+          setZKEVMAssets(assetsResponse.result);
+          console.log(`zkevmAssets: `, assetsResponse)
+        } else {
+          console.log('IMX wallet address is undefined');
+        }
+      } catch (error) {
+        console.error('Error listing ZKEVM assets:', error);
+      } finally {
+        setLoading(false);
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (userProfile) {
+      handleListIMXAssets();
+      handleListZKEVMAssets();
+    }
+  }, [userProfile]);
+
+  const handleAssetSelection = (asset: any) => {
+    setSelectedAssets(prev => 
+      prev.includes(asset) ? prev.filter(a => a !== asset) : [...prev, asset]
+    );
+  };
+
+  const handleBurn = async () => {
+    if (userProfile && burn) {
+      setLoading(true);
+      try {
+        if (imxWalletAddress) {
+          await burn(selectedAssets.map(asset => ({
+            tokenId: asset.token_id,
+            tokenAddress: asset.token_address,
+          })));
+          
+          // Clear selection and refresh asset lists
+          setSelectedAssets([]);
+          await handleListIMXAssets();
+          await handleListZKEVMAssets();
+        } else {
+          console.error('IMX wallet address is undefined');
+        }
+      } catch (error) {
+        console.error('Error burning:', error);
+      } finally {
+        setLoading(false);
+      }
+    }
+  };
+
+  return (
+    <main className="min-h-screen p-8 bg-gray-50">
+      <div className="max-w-6xl mx-auto">
+        <header className="flex justify-between items-center mb-8">
+          <h1 className="text-3xl font-bold text-gray-800">Immutable X to Immutable zkEVM Asset Migrator</h1>
+          {!userProfile ? (
+            <button
+              onClick={handleLogin}
+              className="px-6 py-2 bg-blue-600 text-white rounded-lg hover:bg-blue-700 transition-colors"
+            >
+              Login to Passport
+            </button>
+          ) : (
+            <div className="flex items-center gap-4">
+              <span className="text-gray-600">
+                {imxWalletAddress}
+              </span>
+              <button
+                onClick={logout}
+                className="px-6 py-2 bg-gray-200 text-gray-700 rounded-lg hover:bg-gray-300 transition-colors"
+              >
+                Logout
+              </button>
+            </div>
+          )}
+        </header>
+
+        {userProfile && (
```

Comment:
Generally in JSX/TSX we should pretty much use ternaries rather than a logical AND due to the potential problems that could arise. Article below that explains further into this:

https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx